### DISCUSSION TITLE
librdakafkacpp.dll in librdkafka.redist package does not work correctly

### DISCUSSION THREAD

**Discussion by narensharma88** [edenhill/librdkafka#3296] [Q&A]
Using package librdkafka.redist from nuget, I was not able to run my producer using cpp lib although c producer worked fine.

On first instance of conf->set("bootstrap.servers", brokers, errstr), program crashes as it tries to access invalid memory location.

I had to compile the code including openssl lib then producer worked fine and started pushing messages to broker.

Size of dll in librdkafka.redist is 92 KB
Size of dll after compiling code myself 357 KB

Does anyone else faced same issue? Do we have updated package on nuget that can be used directly?

------- Update --------
I found that it is only happening in debug mode. Using precompiled libs works fine in release mode.

Any idea what could be the issue here?

Compiler version: MSVC++ 14.28 ( where producer is running)
LibrdKafka version: 1.6.1 ( used in producer)
![error](https://user-images.githubusercontent.com/80639526/111900029-29521d80-8a56-11eb-89f8-1d69f6790958.jpg)

**Comment by edenhill** [comment]
I'm not completely sure how the various release/debug modes work with MSVC, but this Stackoverflow answer seems relevant:
https://stackoverflow.com/a/1227918/1821055
